### PR TITLE
Bug fix + Allow Layers to be called  with mask argument

### PR DESCRIPTION
**Pull request by farizrahman4u**
* Current implementation of __call__ does not work for models in which the first layer is also a Sequential.
* Added option to provide mask.

### CODE REVIEW

**Comment by EderSantana** [comment]
:+1: 
@farizrahman4u would you also have time to take a look in this discussion: https://github.com/fchollet/keras/issues/1206

I don't think our masking is working 100% as expected.

**Comment by EderSantana** [comment]
Also, I made you a PR suggesting the same changes for `containers.py`.

**Comment by farizrahman4u** [comment]
@EderSantana What do you think about the fix?

**Comment by EderSantana** [comment]
@farizrahman4u what is the use case? I didn't understand the problem, my bad. Do you think we need another test to cover that?

**Comment by farizrahman4u** [comment]
@EderSantana 

The following code is not working:
```
import keras
from keras.models import Sequential
from keras.layers.core import Dense

m1=Sequential()
m1.add(Dense(input_dim=10,output_dim=10))

m2=Sequential()
m2.add(m1)
m2.add(Dense(5))

X=np.zeros((7,10))
Y=m2(X)
print (Y.eval())
```
Do you get it?

**Comment by EderSantana** [comment]
yeah a little after talking to you I realized that.

**Comment by EderSantana** [comment]
What do you think of the test I PRed? Does it cover that use case?

**Comment by farizrahman4u** [comment]
Yes.it does.But why do you compile the models? We are not training them.

**Comment by EderSantana** [comment]
Just to make sure our process mimics what we would get with the compiled method, which is guaranteed correct by other tests. 

Also, since the tested models aren't large, this doesn't take much more time than compiling just model.get_output(), does it? Do you want to reduce that to 
```python
X = model.get_input()
Y1 = model.get_output()
Y2 = model(X)
F = function([X], [Y1, Y2])
```

Would that be much faster?

**Comment by farizrahman4u** [comment]
Removing the compiling would make it faster..we are unnecessarily calling get_output multiple times..the compile() would call get_output() twice, and then again by __call__

**Comment by farizrahman4u** [comment]
Moved to #1271
